sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/block/CompactBlockStore.kt

```diff
@@ -25,8 +25,9 @@ interface CompactBlockStore {
      * Write the given blocks to this store, which may be anything from an in-memory cache to a DB.
      *
      * @param result the list of compact blocks to persist.
+     * @return Number of blocks that were written.
      */
-    suspend fun write(result: List<CompactFormats.CompactBlock>)
+    suspend fun write(result: Sequence<CompactFormats.CompactBlock>): Int
```

Comment:
Since we're consuming a sequence, we have to get the total count lazily as a return value now.